src/Chapter2.hs

```diff
@@ -842,7 +869,23 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate n _ | n < 0 = []
+rotate _ [] = []
+rotate n l = drop numbersToTake l ++ take numbersToTake (cycle l)
```

Comment:
Alternatively, you can only take the lenght of the initial list elements, out of the list with dropped amount of `div n (length l)` 🙂